homeassistant/components/insteon_plm.py

```diff
@@ -26,95 +26,183 @@
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_PORT): cv.string,
-        vol.Optional(CONF_OVERRIDE, default=[]): cv.ensure_list_csv,
-    })
+        vol.Optional(CONF_OVERRIDE): vol.All(
+            cv.ensure_list_csv, vol.Length(min=1))
```

Comment:
Add a config schema for the override dict. It should validate address, cat, subcat, firmware and product key. Remove the `vol.Length` validator.

Reply:
So if I am going to make this modification I am thinking about adding a check if the old configuration option is still in use (this is the breaking change noted above). If the old configuration is in place, I am considering putting a `persistent_notification` to let the user know that the configuration options have changed. Is this considered the correct use of notifications?

Reply:
You can mark it as deprecated via that validator. It's in the config_validation helper. It will log it. I wouldn't do a persistent notification.

Reply:
Holy crap, this was tough because the documentation for voluptuous is very sparse. But it was definitely the right thing to do. I tested it pretty well so I feel good about it. Thanks for the recommendation.